pex/interpreter.py

```diff
@@ -435,7 +455,7 @@ def create_interpreter(stdout):
         if os.path.isfile(cache_file):
             try:
                 with open(cache_file, "rb") as fp:
-                    return SpawnedJob.completed(create_interpreter(fp.read()))
+                    return SpawnedJob.completed(create_interpreter(fp.read(), check_binary=True))
             except (IOError, OSError, cls.Error, PythonIdentity.Error):
                 safe_rmtree(cache_dir)
```

Comment:
We purge the cache right here. Note the new `check_binary` argument is only passed here (in the try above).